R/stat-summary-2d.R

```diff
@@ -92,31 +92,50 @@ StatSummary2d <- ggproto("StatSummary2d", Stat,
   required_aes = c("x", "y", "z"),
   dropped_aes = "z", # z gets dropped during statistical transformation
 
+  setup_params = function(self, data, params) {
+
+    if (is.character(params$drop)) {
+      params$drop <- !identical(params$drop, "none")
+    }
+
+    params <- fix_bin_params(params, fun = snake_class(self), version = "3.5.2")
+    vars <- c("origin", "binwidth", "breaks", "center", "boundary")
+    params[vars] <- lapply(params[vars], dual_param, default = NULL)
+    params$closed <- dual_param(params$closed, list(x = "right", y = "right"))
+
+    params
+  },
+
   compute_group = function(data, scales, binwidth = NULL, bins = 30,
                            breaks = NULL, origin = NULL, drop = TRUE,
-                           fun = "mean", fun.args = list()) {
-    origin <- dual_param(origin, list(NULL, NULL))
-    binwidth <- dual_param(binwidth, list(NULL, NULL))
-    breaks <- dual_param(breaks, list(NULL, NULL))
+                           fun = "mean", fun.args = list(),
+                           boundary = 0, closed = NULL, center = NULL) {
     bins <- dual_param(bins, list(x = 30, y = 30))
```

Comment:
Ideally this is also done in `setup_params()`, but then the default argument to `compute_group()` should be `bins = list(x = 30, y = 30)`, which causes a test to complain that there is a mismatch between `compute_group()` defaults and `stat_*()` constructor defaults.